### DISCUSSION TITLE
Unexpected Quote Error When Generating XML

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5072] [Q&A]
I am writing a channel that takes in a raw data that is formed like a CCD. I say formed raw data rather than XML because the XML is littered with \ characters that makes Mirth Connect not recognize it as XML.

My solution is to make the incoming message a string, replace all of the bad characters, then create a new XML object. For some reason however my xml is coming back blank and I am getting the following error: TypeError: Open quote is expected for attribute "code" associated with an  element type  "realmCode".

Inputted in to channel:

    <realmCode code=\"US\"/>

'replaced' after the code is ran:

    <realmCode code="US"/>

Source transformer:

    var xmlString = msg.toString()
    var replaced = xmlString.replace(/\\/g, '');
    channelMap.put('replaced', replaced)

    var ccd = new XML(replaced)
    channelMap.put('CCD', ccd)

**Comment by pacmano1** [comment]
Is that XML coming back as response JSON and is simply escaped?

**Comment by tonygermano** [comment]
You have your inbound type set as Raw, I assume? If yes, then `msg` is already a string.

I'd give this a shot just to see if it works.
```js
msg = new XML(JSON.parse('"' + msg + '"'))
```
